samples/bookinfo/src/build-services.sh

```diff
@@ -26,48 +26,75 @@ VERSION=$1
 PREFIX=$2
 SCRIPTDIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 
+# Docker build variables
+ENABLE_MULTIARCH_IMAGES=${ENABLE_MULTIARCH_IMAGES:-"false"}
+
+if [ "${ENABLE_MULTIARCH_IMAGES}" == "true" ]; then
+  PLATFORMS="linux/arm64,linux/amd64"	
+  DOCKER_BUILD_ARGS="docker buildx build --platform ${PLATFORMS} --push"
+  # Install QEMU emulators
+  docker run --rm --privileged tonistiigi/binfmt --install all
+  docker buildx rm multi-builder || :
+  docker buildx create --use --name multi-builder --platform ${PLATFORMS}
+  docker buildx use multi-builder
+else
+  DOCKER_BUILD_ARGS="docker build"	
+fi
+
 pushd "$SCRIPTDIR/productpage"
-  docker build --pull -t "${PREFIX}/examples-bookinfo-productpage-v1:${VERSION}" -t "${PREFIX}/examples-bookinfo-productpage-v1:latest" .
+  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-productpage-v1:${VERSION}" -t "${PREFIX}/examples-bookinfo-productpage-v1:latest" .
   #flooding
-  docker build --pull -t "${PREFIX}/examples-bookinfo-productpage-v-flooding:${VERSION}" -t "${PREFIX}/examples-bookinfo-productpage-v-flooding:latest" --build-arg flood_factor=100 .
+  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-productpage-v-flooding:${VERSION}" -t "${PREFIX}/examples-bookinfo-productpage-v-flooding:latest" --build-arg flood_factor=100 .
 popd
 
 pushd "$SCRIPTDIR/details"
   #plain build -- no calling external book service to fetch topics
-  docker build --pull -t "${PREFIX}/examples-bookinfo-details-v1:${VERSION}" -t "${PREFIX}/examples-bookinfo-details-v1:latest" --build-arg service_version=v1 .
+  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-details-v1:${VERSION}" -t "${PREFIX}/examples-bookinfo-details-v1:latest" --build-arg service_version=v1 .
   #with calling external book service to fetch topic for the book
-  docker build --pull -t "${PREFIX}/examples-bookinfo-details-v2:${VERSION}" -t "${PREFIX}/examples-bookinfo-details-v2:latest" --build-arg service_version=v2 \
+  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-details-v2:${VERSION}" -t "${PREFIX}/examples-bookinfo-details-v2:latest" --build-arg service_version=v2 \
 	 --build-arg enable_external_book_service=true .
 popd
 
+
 pushd "$SCRIPTDIR/reviews"
   #java build the app.
   docker run --rm -u root -v "$(pwd)":/home/gradle/project -w /home/gradle/project gradle:4.8.1 gradle clean build
+  
+  # Temporarily clone https://github.com/OpenLiberty/ci.docker
+  # to allow this project to build multiarch(linux/arm64) images not provided by the OpenLiberty CI
+  # Remove this workaround once https://github.com/OpenLiberty/ci.docker/issues/241 is fixed
+  git clone https://github.com/OpenLiberty/ci.docker.git openliberty-base
+  pushd openliberty-base
+    ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/websphere-liberty:22.0.0.8-full-java8-ibmjava" -f releases/22.0.0.8/full/Dockerfile.ubuntu.openjdk8 releases/22.0.0.8/full 
```

Comment:
we could just build it once and push to gcr.io/istio-testing?

Reply:
sure, if you have the super powers to build and push, can you edit that in a follow up PR ?